### PR TITLE
MM-38465 Add Discard Modal for Company Information Edit

### PR DESCRIPTION
**Pull request by ConorMacpherson**
#### Summary

When editing company information, the standard discard modal is not presented when navigating to another page.
This change ensures the modal appears when any field is modified.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-38465

#### Screenshots

Screen Shot 2023-01-24 at 10 53 02 AM

#### Test Plan

1. Go to System Console ➜ Billing & Account ➜ Company Information
2. Add a new address without clicking the Save Info button
3. Click the back arrow to exit
**Expected**: The standard System Console Discard Changes modal prevents me from leaving (see screenshot) 

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Comment by ConorMacpherson** [comment]
@Afsoon @AGMETEOR Friendly nudge on these when you two are online again :)

**Review by Afsoon — APPROVED** [review]

**Comment by ConorMacpherson** [comment]
@JulienTant Any chance I could redeem that one free code review here? :)

File: components/admin_console/billing/company_info_edit.tsx [hunk 1]
```diff
@@ -54,12 +57,14 @@ const CompanyInfoEdit: React.FC<Props> = () => {
...
             setStateFunc(event.target.value);
             setValidation();
+            setContentChanged(true);
         };
     };
 
     const updateNumEmployees = (event: React.ChangeEvent<HTMLInputElement>) => {
         if (event.target.value) {
             setNumEmployees(parseInt(event.target.value, 10));
+            setContentChanged(true);
```
Comment (JulienTant) [inline components/admin_console/billing/company_info_edit.tsx:67]: from a purely logic perspective, I would expect `setContentChanged` to be true even if the input is empty/wrong/malformed.

**Review by JulienTant — APPROVED** [review]
LGTM, left a nit that's probably irrelevant.

File: components/admin_console/billing/company_info_edit.tsx (hunk 1 above)
Comment (ConorMacpherson) [reply components/admin_console/billing/company_info_edit.tsx]: Definitely! Nice catch :)

**Review by stevemudie — APPROVED** [review]
Looks good @ConorMacpherson!

**Pull request merged**